### ISSUE TITLE
$$ doesn't work

### ISSUE THREAD

**Issue by Jojo-lan** [aarkue/tiptap-math-extension#9]
When I print $$ and then insert the formula between them, it doesn't work because the input rule of tiptap doesn't support it. Would you please give me some ideas to fix it? Thank you.

**Comment by aarkue** [comment]
Hi @Jojo-lan,
could you provide the latex which is not working or maybe even a minimal example?

**Comment by Jojo-lan** [comment]
For example, when I print $$ and then print sin(x) between them, the sin(x) won't be translated into latex formula, I think that is because the inputRule can't deal with this condition.

**Comment by Jojo-lan** [comment]
Thank you so much! Now I got that inputRule can't deal with this condition. I wonder if there is anyway that I can deal with it. I create a custom node and use a method like tiptap/mathematics-pro-extenson to convert the text between $ into that node, when I do this $$ -> $s$ -> $si$ -> $sin(x)$ , the text is translated into my custom node immediately in the second step  $s$, so I can't input the following words. In the pro-extension, the text is not converted into node so I can input the full word. Is it possible to convert text into node after inputing the whole word like this: $$ -> $s$ -> $sin(x)$->sin(x)[custom node]

**Comment by aarkue** [comment]
Yeah, the pro extension (I think) uses more underlying ProseMirror API surface for this to work.

If this is required for your use case, you could maybe try https://github.com/benrbray/prosemirror-math which uses a similar approach, as far as I know. It should be easy to also use this in TipTap (as it is built on top of prosemirror). However, I never tried it, as I personally prefer the approach of this package (i.e., LaTeX as full nodes and not text content).

Cheers :)

**Comment by Jojo-lan** [comment]
Thank you so much, I will have a try😊

**Comment by p-c-mo** [comment]
Hey, @Jojo-lan did you ever pursue this?
